### PR TITLE
Add consistent-component-name rule

### LINKED ISSUE #3: Rule idea: Warn when the component class name doesn't match the file name
When you want to create a new component, it often happens that you duplicate an existing component or you copy-paste a component class to get started. We should add a new rule to ensure that component class name matches the file name.

In the `ns/listItem/listItem.js` file:

```js
import { LightningElement } from 'lwc';
export default class Foo extends LightningElement {}
//                   ^ The component class "Foo" doesn't match the filename.
```

### PR DESCRIPTION
**Pull request by pmdartus**
Fix #3

### CODE REVIEW

**Comment by pmdartus** [comment]
I don't feel really inspired by the rule name nor by the error messages, I am opened to any better suggestion.

File: test/lib/rules/consistent-component-name.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            code: `export class Foo extends LightningElement {}`,
+            filename: 'foo.js',
+        },
+        {
+            code: `export class ComplexName extends LightningElement {}`,
+            filename: 'complexName.js',
+        },
+    ],
+    invalid: [
```
Comment (trevor-bliss) [inline test/lib/rules/consistent-component-name.js]: How about a test case where the casing in the class name is off but the characters are correct?
```
code: `export class complexName extends LightningElement {}`,
filename: 'complexName.js'
```

File: test/lib/rules/consistent-component-name.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            code: `export default class extends LightningElement {}`,
+            filename: 'foo.js',
+            errors: [
+                {
+                    type: 'ClassDeclaration',
+                    column: 16,
+                    message: 'Lightning component class should be named "Foo".',
+                },
+            ],
+            output: 'export default class Foo extends LightningElement {}',
```
Comment (trevor-bliss) [inline test/lib/rules/consistent-component-name.js]: The `RuleTester` we import from eslint automatically verifies the code fix here, correct? Looking through the source I believe it does but wanted to double check.

File: README.md [hunk 3]
```diff
@@ -32,27 +32,28 @@ For more details about configuration please refer to the dedicated section in th
...
-| [lwc/valid-wire](./docs/rules/valid-wire.md)               | validate `wire` decorator usage          |
-| [lwc/no-document-query](./docs/rules/no-document-query.md) | Disallow DOM query at the document level |
+| Rule ID                                                                    | Description                                 | Fixable |
+| -------------------------------------------------------------------------- | ------------------------------------------- | ------- |
+| [lwc/no-deprecated](./docs/rules/no-deprecated.md)                         | disallow usage of deprecated LWC APIs       |         |
+| [lwc/valid-api](./docs/rules/valid-api.md)                                 | validate `api` decorator usage              |         |
+| [lwc/valid-track](./docs/rules/valid-track.md)                             | validate `track` decorator usage            |         |
+| [lwc/valid-wire](./docs/rules/valid-wire.md)                               | validate `wire` decorator usage             |         |
+| [lwc/no-document-query](./docs/rules/no-document-query.md)                 | Disallow DOM query at the document level    | ✅      |
```
[the remark this answers is not in the document]
Comment (pmdartus) [reply README.md]: Good catch

File: test/lib/rules/consistent-component-name.js (hunk 2 above)
Comment (pmdartus) [reply test/lib/rules/consistent-component-name.js]: Yes, it's the case. If the `output` is not the generated output by the fixer the test fails.

**Comment by pmdartus** [comment]
@trevor-bliss Ready for review.

**Pull request merged**